### DISCUSSION TITLE
mini visits - only track files in the current dir or its children

### DISCUSSION THREAD

**Discussion by beaumccartney** [nvim-mini/mini.nvim#621] [Q&A] [answered]
I tend to use goto-definition a lot to look at system libraries and things - what is the simplest way to have mini.visits only track files that are children of PWD and not ignored in git?

**Comment by echasnovski** [comment, marked as answer]
There are at least two approaches here:
- Create an autocommand on `BufAdd` which sets `vim.b.minivisits_disable` if it doesn't fit requirements for being tracked. This is doable and does directly what you want.
- A slightly alternative solution would be to define your own `config.store.normalize` function which removes unwanted path entries. This looks like a slightly cleaner and more performant approach, but it means that unwanted paths will be in index before normalization is done (during writing to disk or manually calling `normalize()`). Here is a snippet which removes non-children entries:

```lua
local visits = require('mini.visits')
local default_normalize = visits.gen_normalize.default()
local normalize = function(index)
  for cwd, cwd_tbl in pairs(index) do
    for path, _ in pairs(cwd_tbl) do
      if not vim.startswith(path, cwd) then cwd_tbl[path] = nil end
    end
  end

  return default_normalize(index)
end

visits.setup({ store = { normalize = normalize } })
```
